src/main/java/am/ik/yavi/constraint/EnumConstraint.java

```diff
@@ -0,0 +1,23 @@
+package am.ik.yavi.constraint;
+
+import am.ik.yavi.constraint.base.ConstraintBase;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+/**
+ * class for enum constraints
+ * @since 0.14.0
+ */
+public class EnumConstraint<T, E extends Enum>
+		extends ConstraintBase<T, E, EnumConstraint<T, E>> {
+
+	@Override
+	public EnumConstraint<T, E> cast() {
+		return this;
+	}
+
+	public EnumConstraint<T, E> oneOf(E... values) {
+		return super.oneOf(Arrays.stream(values).collect(Collectors.toList()));
```

Comment:
Why not EnumSet as same as the initial pull request?

Reply:
> Why not EnumSet as same as the initial pull request?

makes sense, converting the array to EnumSet would ensure that the collections doesnt have two enums of the same values.

is this what you meant?:
```
	public EnumConstraint<T, E> oneOf(E... values) {
		EnumSet enumSet = EnumSet.copyOf(Arrays.asList(values));

		return super.oneOf(enumSet);
	}
```